source/slang/slang-emit-source-writer.cpp

```diff
@@ -239,35 +239,197 @@ void SourceWriter::emit(double value)
     // 3) We need to be sure that floating-point literals specified by the user will
     //    "round-trip" and turn into the same value when parsed back in. This means
     //    that we need to print a reasonable number of digits of precision.
-    //
-    // For right now, the easiest option that can balance these is to use
-    // the C++ standard library `iostream`s, because they support an explicit locale,
-    // and can (hopefully) print floating-point numbers accurately.
-    //
-    // Eventually, the right move here would be to implement proper floating-point
-    // number formatting ourselves, but that would require extensive testing to
-    // make sure we get it right.
-
-    std::ostringstream stream;
-    stream.imbue(std::locale::classic());
-    stream.setf(std::ios::fixed, std::ios::floatfield);
-    stream.precision(20);
-    stream << value;
-    auto str = stream.str();
-    auto slice = UnownedStringSlice(str.c_str());
-    // Remove redundant trailing 0s.
-    if (slice.end() > slice.begin())
+
+    char result[30];
+    char *next = result;
+    double valueAbs = value;
+
+    // Handle negative sign
+    const long long* d2l = reinterpret_cast<long long*>(&value);
+    const bool negative = (*d2l < 0);
```

Comment:
Can this just be `sign(value)`?

Reply:
The tricky part is most of attempt to get the sign-ness will fail if the value is NaN.
I see that there is a C++11 function for this, signbit.
I will use it https://en.cppreference.com/w/cpp/numeric/math/signbit